### DISCUSSION TITLE
Need help applying CSS to my website

### DISCUSSION THREAD

**Discussion by Degeben** [community/community#44522] [Other Feature Feedback, Questions, & Ideas] [answered]
GitHub Pages CSS 

Question

### Body

Hi, new to GitHub and I'm a little confused as to what as going wrong with my GitHub Pages. I have created a repo with all of my website files, I've followed the docs to get my page working, but for some reason it seems to just be my basic index.html without all my CSS works. I found a forum post that had a similar issue, I tried following the help suggested there but couldn't seem to get it to work. Just not sure if I'm missing something or what, any help would be great thanks! 

Here's a link to my repo:
https://github.com/Degeben/Degeben

**Comment by radar-blip** [comment, marked as answer]
You could try bringing your style.css out into the main part of your repo along with your index.html
and change the link to: `<link href="style.css" rel="stylesheet">`
That should work.

**Reply by RafaelNevesdeOliveira** [reply]
Can you share your index.html?

**Reply by Degeben** [reply]
It should be in my repo I linked, but I tried moving everything out of folders and into the base repo along with my index.html but it still doesn't seem to be working. I went through and updated my index.html to ref style.css like you mentioned as well but still nothing.

**Comment by idadwind1** [comment]
![screenshot 2023-01-17 121426](https://user-images.githubusercontent.com/89760671/212808671-42281da9-80d7-4869-aee2-4d1519ed4be9.png)
Um, seems like your files did not upload successfully except index.html.
![screenshot 2023-01-17 121759](https://user-images.githubusercontent.com/89760671/212809002-cd471bd7-b1dc-4a43-9fac-abc5b450c740.png)
Did you try uploading them again?

**Reply by Degeben** [reply]
Yeah I'm not sure, I can get it to load at https://degeben.github.io/Degeben/ but without the /Degeben it doesn't work. Once I can get it to load however; it is still without my CSS, new to GitHub and GitHub Pages so still figuring this out!

**Reply by idadwind1** [reply]
I found a solution, it may help:
create a file ".nojekyll" at the root to disable Jekyll.

**Reply by idadwind1** [reply]
If this doesn't work, try replacing all relative paths in index.html with absolute path

**Reply by Degeben** [reply]
Ok I think I created the .nojekyll correctly, I will go ahead and change any relative paths to absolute paths in my index.html! Thank you for the responses :)

**Reply by Degeben** [reply]
I'm still getting 404 so not sure, I'm thinking I'm missing something obvious at this point

**Reply by idadwind1** [reply]
OK, I got a trump card
1. Save your website to local
2. delete this repository
3. Create a new repository called "Degeben.github.io"
5. upload your website to this new repository
6. Now visit degeben.github.io

**Reply by Degeben** [reply]
I will give that a go thank you so much for your help!

**Reply by idadwind1** [reply]
you're welcome

**Reply by Degeben** [reply]
I did that and now it just loads indefinitely so not sure what's up haha, thanks for the help anyways!

**Reply by radar-blip** [reply]
Something to do with the .nojekyll ? Wasn't it working before that was added?

**Reply by idadwind1** [reply]
I think it may in relation to the absolute path, did you change it correctly when you move to another repository or path?

**Reply by Degeben** [reply]
I deleted .nojekyll and still nothing, but I believe everything is correct. I have the style.css in the main repo alongside my index.html so I kept it as the absolute path. I will try changing the other CSS file paths to match since they are in folders now instead of just being  out in the open.

**Comment by radar-blip** [comment]
When I view your index.html in your repo, its like this:   `<link href="style.css" rel="stylesheet">`

But viewing page source from the actual deployed site, its showing: `<link href="assets/css/style.css" rel="stylesheet">`

I dont know why this is, unless your index.html isn't getting updated correctly.

**Reply by Degeben** [reply]
Is there any reason for that? I thought I have updated and pushed all updated I've made? Does it just take time to update?

**Reply by radar-blip** [reply]
I'm not sure, I'm kinda baffled by it myself. 😕

**Reply by Degeben** [reply]
No worries Im going to delete and retry! Thank you for your help!

**Reply by radar-blip** [reply]
No problem. Good luck!

**Comment by mo078-lang** [comment]
> You could try bringing your style.css out into the main part of your repo along with your index.html
> 
> and change the link to: `<link href="style.css" rel="stylesheet">`
> 
> That should work.
> 
>
